File: sources/pharmacy_delivery/lib/class/Drugstore.dart
import 'Owner.dart';

class Drugstore {
  final String? drugstoreID;
  final Owner? owner;
  final String? drugstoreName;
  final String? drugstoreAddress;
  final String? drugstoreTel;
  final String? drugstoreStatus;
  final String? drugstoreImg;

/*
  Drugstore( this.drugstoreID, this.owner, this.drugstoreName,
      this.drugstoreAddress, this.drugstoreTel, this.drugstoreStatus,
      this.drugstoreImg);

 */
  Drugstore({
     this.drugstoreID,
     this.owner,
     this.drugstoreName,
     this.drugstoreAddress,
     this.drugstoreTel,
     this.drugstoreStatus,
     this.drugstoreImg,
  });

  factory Drugstore.fromJson(Map<String, dynamic> json) {
    return Drugstore(
        drugstoreID: json['drugstoreID'] == null ? null :json["drugstoreID"] ,
        owner: json['owner'] == null ? null :Owner.fromJson(json['owner']),
        drugstoreName: json['drugstoreName'] == null ? null :json["drugstoreName"] ,
        drugstoreAddress: json['drugstoreAddress'] == null ? null :json["drugstoreAddress"] ,
        drugstoreTel: json['drugstoreTel'] == null ? null :json["drugstoreTel"] ,
        drugstoreStatus: json['drugstoreStatus']== null ? null :json["drugstoreStatus"] ,
        drugstoreImg: json['drugstoreImg'] == null ? null :json["drugstoreImg"] ,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      "drugstoreID": this.drugstoreID == null ? null : this.drugstoreID,
      "owner":this.owner == null ? null :  this.owner?.toJson(),
      "drugstoreName": this.drugstoreName== null ? null : this.drugstoreName,
      "drugstoreAddress": this.drugstoreAddress== null ? null : this.drugstoreAddress,
      "drugstoreTel": this.drugstoreTel== null ? null : this.drugstoreTel,
      "drugstoreStatus": this.drugstoreStatus== null ? null : this.drugstoreStatus,
      "drugstoreImg": this.drugstoreImg== null ? null : this.drugstoreImg,
    };
  }

  Map<String, dynamic> toJsonDrugstoreID() {
    return {
      "drugstoreID": this.drugstoreID == null ? null : this.drugstoreID,
    };
  }

}

File: sources/WebService/PharmacyDelivery/src/bean/Drugstore.java
package bean;
import javax.persistence.*;

@Entity
@Table(name="Drugstore")

public class Drugstore {
	@Id
	@Column(name="drugstoreID")
	private String drugstoreID;
	
	@ManyToOne(cascade = CascadeType.ALL)
	@JoinColumn(name="ownerid",nullable=false)
	private Owner owner ;
	
	@Column(name="drugstoreName")
	private String drugstoreName;
	
	@Column(name="drugstoreAddress")
	private String drugstoreAddress;
	
	@Column(name="drugstoreTel")
	private String drugstoreTel;
	
	
	@Column(name="drugstoreStatus")
	private String drugstoreStatus;

	
	@Column(name="drugstoreImg")
	private String drugstoreImg;
	
	public Drugstore() {
	}
	
	public Drugstore(String drugstoreID) {
		this.drugstoreID = drugstoreID;
	}

	public Drugstore(String drugstoreID, Owner owner, String drugstoreName, String drugstoreAddress,
			String drugstoreTel, String drugstoreStatus, String drugstoreImg) {
		super();
		this.drugstoreID = drugstoreID;
		this.owner = owner;
		this.drugstoreName = drugstoreName;
		this.drugstoreAddress = drugstoreAddress;
		this.drugstoreTel = drugstoreTel;
		this.drugstoreStatus = drugstoreStatus;
		this.drugstoreImg = drugstoreImg;
	}

	public String getDrugstoreID() {
		return drugstoreID;
	}

	public void setDrugstoreID(String drugstoreID) {
		this.drugstoreID = drugstoreID;
	}

	public Owner getOwner() {
		return owner;
	}

	public void setOwner(Owner owner) {
		this.owner = owner;
	}

	public String getDrugstoreName() {
		return drugstoreName;
	}

	public void setDrugstoreName(String drugstoreName) {
		this.drugstoreName = drugstoreName;
	}

	public String getDrugstoreAddress() {
		return drugstoreAddress;
	}

	public void setDrugstoreAddress(String drugstoreAddress) {
		this.drugstoreAddress = drugstoreAddress;
	}

	public String getDrugstoreTel() {
		return drugstoreTel;
	}

	public void setDrugstoreTel(String drugstoreTel) {
		this.drugstoreTel = drugstoreTel;
	}

	public String getDrugstoreStatus() {
		return drugstoreStatus;
	}

	public void setDrugstoreStatus(String drugstoreStatus) {
		this.drugstoreStatus = drugstoreStatus;
	}

	public String getDrugstoreImg() {
		return drugstoreImg;
	}

	public void setDrugstoreImg(String drugstoreImg) {
		this.drugstoreImg = drugstoreImg;
	}


	
	
	
	
}
